### DISCUSSION TITLE
Kubernetess Secrets retrieving

### DISCUSSION THREAD

**Discussion by Nicknameee** [apache/camel-karavan#1317] [Q&A] [answered]
### Describe the solution you'd like

Good afternoon, this issue regarding best approach to work with secrets in Kubernetess using Apache Camel Karavan UI

Currently we using two options: integrated Kubernetess selector in UI itself, Java-based processors
1)
![image](https://github.com/apache/camel-karavan/assets/98988322/b2f1d8f6-b792-49cd-a496-05a7273b85e0)
2)
![image](https://github.com/apache/camel-karavan/assets/98988322/15d57986-cfa0-4da8-9ea0-907433e2e892)
Both options works, but we would like to collect reasonable advice or hint for possible better approach existence.
For example, using secrets value in SetHeader component
![image](https://github.com/apache/camel-karavan/assets/98988322/2b6e46aa-5797-40ef-99ed-53080184b5bf)
Is there any way to use regular expressions or some other solution to avoid using separate Java-based processor that works directly with Exchange
For instance, using same pattern from components that support direct secret fetching {{secret:SECRET_NAME/KEY}}
Best regards

### Variant

Web Application

### Container Management (if applicable)

Kubernetes

**Comment by davsclaus** [comment, marked as answer]
In karavan use the out of the box support with `{{secret:xxx}}` style.

If you need to inject into Java code, you can use `CamelContext / Exchange` API or dependency injection with `@PropertyInject` etc. which is more advanced and what you would do with Spring Boot Java coding etc.

**Reply by Nicknameee** [reply]
Good afternoon, we got onto desired results using SPeL language in combo with `secret` style
Thank you for your assistance
![image](https://github.com/apache/camel-karavan/assets/98988322/eaf0eaeb-7d5f-4503-bd1c-27401fe9c72c)
